content/en/guide/v10/forms.md

```diff
@@ -95,20 +97,39 @@ class MySelect extends Component {
 }
 ```
 
-## Checkboxes & Radio Buttons
+## Checkbox and Radio Button
+
+In HTML, browser toggles and checks checkbox and radio button. In Controlled Component, you need to implement toggle and check. Therefore, checkbox and radio button (`<input type="checkbox|radio">`) can initially cause confusion When you build Form in Controlled Component.
 
-Checkboxes and radio buttons (`<input type="checkbox|radio">`) can initially cause confusion when building controlled forms. This is because in an uncontrolled environment, we would typically allow the browser to "toggle" or "check" a checkbox or radio button for us, listening for a change event and reacting to the new value.  However, this technique does not transition well into a world view where the UI is always updated automatically in response to state and prop changes.
+Say we listen for a "change" event on a checkbox in Controlled Component, which is fired when the checkbox is checked or unchecked by the user. Form component set `checked` value in the Event object received from the checkbox to `state` in `change` event handler. This triggers rerendering component. `checked` value in `state` is reflected in the checkbox by this rerendering.
 
-> **Walk-Through:** Say we listen for a "change" event on a checkbox, which is fired when the checkbox is checked or unchecked by the user.  In our change event handler, we set a value in `state` to the new value received from the checkbox.  Doing so will trigger a re-render of our component, which will re-assign the value of the checkbox to the value from state.  This is unnecessary, because we just asked the DOM for a value but then told it to render again with whatever value we wanted.
```

Comment:
Does " This is unnecessary, because we just asked the DOM for a value but then told it to render again with whatever value we wanted." mean the following?

```jsx
class MyForm extends Component {
  toggle = event => {
      const checked = event.target.checked;
      this.setState({ checked });
  };
  render(_, { checked }) {
    return (
      <label>
        <input
          type="checkbox"
          onChange={this.toggle}
        />
        foo
      </label>
    );
  }
}
```

I may be misunderstanding.

Reply:
This sentence is difficult for me.
I think "we just asked the DOM for a value" is "getting checkbox's value".
What exactly does "whatever value we wanted" mean?
"This is unnecessary, because we just asked the DOM for a value but then told it to render again with whatever value we wanted."